### PR TITLE
Fixes #10676: Refactoring and warning messages improvement for ConfigNodeClient

### LINKED ISSUE #10676: ConfigNodeClient’s exception log needs improvement
Currently, whenever a DataNode sends a request to a ConfigNode and an TException is caught due to any reason, an "Failed to connect to ConfigNode" exception is logged in the system. 
However, in reality, the exception may not necessarily be a connection issue. 
This could be misleading and cause confusion during issue investigation, as developers might falsely assume that the issue is related to connecting to the ConfigNode, when it may not be the case.

### PR DESCRIPTION
**Pull request by liyuheng55555**
resolve this issue: https://github.com/apache/iotdb/issues/10676
doc: https://apache-iotdb.feishu.cn/docx/Gd1SdLZaIoYBsSxJ1MTc6AN5niZ

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java [hunk 1]
```diff
@@ -1065,150 +634,50 @@ public TSStatus setDataNodeStatus(TSetDataNodeStatusReq req) throws TException {
...
   public TShowDataNodesResp showDataNodes() throws TException {
-    for (int i = 0; i < RETRY_NUM; i++) {
-      try {
-        TShowDataNodesResp showDataNodesResp = client.showDataNodes();
-        showDataNodesResp.setStatus(showDataNodesResp.getStatus());
-        if (!updateConfigNodeLeader(showDataNodesResp.getStatus())) {
+    return executeRemoteCallWithRetry(
+        () -> {
+          TShowDataNodesResp showDataNodesResp = client.showDataNodes();
+          showDataNodesResp.setStatus(showDataNodesResp.getStatus());
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java:658]: seems redundant

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java [hunk 2]
```diff
@@ -315,27 +315,48 @@ private boolean updateConfigNodeLeader(TSStatus status) {
...
       try {
-        TSystemConfigurationResp resp = client.getSystemConfiguration();
-        if (!updateConfigNodeLeader(resp.status)) {
-          return resp;
+        T result = call.execute();
+        if (!check.apply(result)) {
+          return result;
         }
       } catch (TException e) {
         logger.warn(
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java:336]: maybe we should override `void warn(String var1, Throwable var2);`

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java (hunk 1 above)
Comment (liyuheng55555) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java]: deleted.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java (hunk 2 above)
Comment (liyuheng55555) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java]: done

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**